Simplified/NYPLRootTabBarController.m

```diff
@@ -50,7 +50,9 @@ - (instancetype)init
   self.catalogNavigationController = [[NYPLCatalogNavigationController alloc] init];
   self.myBooksNavigationController = [[NYPLMyBooksNavigationController alloc] init];
   self.holdsNavigationController = [[NYPLHoldsNavigationController alloc] init];
-  self.settingsSplitViewController = [[NYPLSettingsSplitViewController alloc] init];
+  self.settingsSplitViewController = [[NYPLSettingsSplitViewController alloc]
+                                      initWithCurrentLibraryAccountProvider:
+                                      AccountsManager.shared];
```

Comment:
@ErnestFan we talked about this some time ago. As you can see here, the dependency on `AccountsManager` that the `NYPLSettingsSplitViewController` would have (in order to get the current Account's URLs) is now explicit, i.e. expressed outside of the class. However the NYPLSettingsSplitViewController interface only sees a protocol (NYPLCurrentLibraryAccountProvider). 

If instead we hard-coded the usage of `AccountsManager` inside the `NYPLSettingsSplitViewController` implementation (as often do everywhere else in the code base!), the 2 classes would be very tightly coupled: you could not even compile the SplitVC without the AccountsManager. This dependency would be implicit, in the sense that by looking at only the interface of `NYPLSettingsSplitViewController` you could not know that its implementation depends on `AccountsManager`.

By breaking this coupling via a protocol the SplitVC doesn't know about the `AccountsManager`. It only knows about the `NYPLCurrentLibraryAccountProvider` protocol (implemented by `AccountsManager`). This is the key point. This would allow us to write unit tests for SplitVC without needing to set up a huge class like `AccountsManager`.

This case here is not the best example (unit testing SplitVC might be good, sure, but there are other areas that need testing more urgently), but it's an example of how I would like us to move forward in the future, especially when dealing with singletons.